### PR TITLE
EVG-14308 Add execution option to /tasks API endpoint

### PR DESCRIPTION
**Pull request by bynn**
`http://localhost:9090/rest/v2/tasks/evergreen_ubuntu1604_test_service_patch_5e823e1f28baeaa22ae00823d83e03082cd148ab_5e4ff3abe3c3317e352062e4_20_02_21_15_13_48?execution=0`
image

`http://localhost:9090/rest/v2/tasks/evergreen_ubuntu1604_test_service_patch_5e823e1f28baeaa22ae00823d83e03082cd148ab_5e4ff3abe3c3317e352062e4_20_02_21_15_13_48?execution=1`
image

### CODE REVIEW

File: rest/data/task.go [hunk 1]
```diff
@@ -259,6 +273,16 @@ func (mtc *MockTaskConnector) FindTaskById(taskId string) (*task.Task, error) {
...
+func (mtc *MockTaskConnector) FindTaskByIdAndExecution(taskId string, execution int) (*task.Task, error) {
+	for _, t := range mtc.CachedTasks {
+		if t.Id == taskId && t.Execution == execution {
+			return &t, mtc.StoredError
+		}
+	}
+	return nil, mtc.StoredError
+}
+
 // FindTasksBytaskId
```
Comment (bynn) [inline rest/data/task.go:286]: why is this here?

**Review by malikchaya2 — COMMENTED** [review]
Can you please add unit tests?

File: rest/data/task.go [hunk 2]
```diff
@@ -37,6 +37,20 @@ func (tc *DBTaskConnector) FindTaskById(taskId string) (*task.Task, error) {
...
 
+func (tc *DBTaskConnector) FindTaskByIdAndExecution(taskId string, execution int) (*task.Task, error) {
+	t, err := task.FindByIdExecution(taskId, &execution)
+	if err != nil {
+		return nil, err
+	}
+	if t == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task with id %s not found", taskId),
```
Comment (ablack12) [inline rest/data/task.go:48]: nit: wrap formatting with `` for readability (i.e. `%s`)

File: rest/data/task.go (hunk 1 above)
Comment (ablack12) [reply rest/data/task.go:286]: definitely leftover, you can delete

File: rest/data/task.go [hunk 3]
```diff
@@ -259,6 +273,16 @@ func (mtc *MockTaskConnector) FindTaskById(taskId string) (*task.Task, error) {
 	return nil, mtc.StoredError
 }
 
+// FindTaskByIdAndExecution
```
Comment (ablack12) [inline rest/data/task.go:276]: This comment isn't really necessary

File: rest/route/task.go [hunk 4]
```diff
@@ -43,13 +45,43 @@ func (tgh *taskGetHandler) Factory() gimlet.RouteHandler {
 func (tgh *taskGetHandler) Parse(ctx context.Context, r *http.Request) error {
 	tgh.taskID = gimlet.GetVars(r)["task_id"]
 	_, tgh.fetchAllExecutions = r.URL.Query()["fetch_all_executions"]
+	execution := r.URL.Query().Get("execution")
+
+	if execution != "" && tgh.fetchAllExecutions == true {
+		return gimlet.ErrorResponse{
+			Message:    "fetchAllExecutions=true cannot be combined with execution={task_execution}",
```
Comment (ablack12) [inline rest/route/task.go:52]: users type this as "fetch_all_executions", not fetchAllExecutions

File: rest/route/task.go [hunk 5]
```diff
@@ -43,13 +45,43 @@ func (tgh *taskGetHandler) Factory() gimlet.RouteHandler {
 func (tgh *taskGetHandler) Parse(ctx context.Context, r *http.Request) error {
 	tgh.taskID = gimlet.GetVars(r)["task_id"]
 	_, tgh.fetchAllExecutions = r.URL.Query()["fetch_all_executions"]
+	execution := r.URL.Query().Get("execution")
+
+	if execution != "" && tgh.fetchAllExecutions == true {
```
Comment (ablack12) [inline rest/route/task.go:50]: you don't need to add `== true` -- this is implied because fetchAllExecutions is a boolean

File: rest/data/task.go (hunk 2 above)
Comment (ablack12) [reply rest/data/task.go:48]: Some of these comments say done but I'm thinking you forgot to push them; I resolved the ones that actually look done.

Comment (bynn) [reply rest/data/task.go]: I think it's because i used single quotes instead of a backtick 
fixed

File: rest/data/task.go (hunk 3 above)
Comment (bynn) [reply rest/data/task.go]: fixed

File: rest/data/task.go (hunk 1 above)
Comment (bynn) [reply rest/data/task.go]: fixed

**Review by bynn — COMMENTED** [review]
I'm not sure why the quote thing didn't work the first time but I just deleted it and wrote it again.

Some of the deletions just wasnt saved locally until now!

**Review by ablack12 — APPROVED** [review]
